### DISCUSSION TITLE
Version and Language pickers in doc portal

### DISCUSSION THREAD

**Discussion by syeda-git** [squidfunk/mkdocs-material#3614] [Q&A]
Hi,
I have both Version and Language pickers in the top bar
https://syeda-git.github.io/mkdocs-sandbox/9.5/

However, when i switch the language, the Version picker disappears. What could be the reason and how can it be fixed?
Also, when switching the language, instead of changing the language of the current (9.5) version, it opens the language from 10.0. Any idea, what issue is causing this problem?

Thanks...
Syed

**Comment by squidfunk** [comment]
You've added the versions to the language picker in `mkdocs.yml`, which is why versions are added to the links:
https://github.com/syeda-git/mkdocs-sandbox/blob/main/mkdocs.yml

Furthermore, it appears that you're using a [self-baked solution for versioning](https://github.com/syeda-git/mkdocs-sandbox/blob/main/versions.json), which we cannot offer support for. A better idea is to use `mike`, which we officially support.

**Reply by syeda-git** [reply]
Do you mean that the paths should not include the version numbers? Something like this:

```  
     - name: English
          link: /mkdocs-sandbox/
          link: /mkdocs-sandbox/
          lang: en
  
      - name: Francais
          link: /mkdocs-sandbox/
          link: /mkdocs-sandbox/
          lang: fr 
```

Ignore the versions.json file in the main branch. The real [versions.json](https://github.com/syeda-git/mkdocs-sandbox/blob/version/versions.json) is in the version branch. I intend to use `mike` for versioning.

**Reply by squidfunk** [reply]
Yes, if you hardcode the version in the language selector entry, it will always link to that version. There's no magic involved.
